packages/project-disaster-trail/src/components/Game/TaskScreen/index.js

```diff
@@ -1,40 +1,240 @@
+      chapterTimer.stop();
     };
-  }, []);
+  }, [chapterTimer]);
+
+  useEffect(() => {
+    switch (action) {
+      case ACTIONS.SOLVING:
+        if (prevActiveTask && !activeTask) {
+          setVotingComplete(false);
+          setAction(ACTIONS.VOTING);
+        }
+        break;
+      case ACTIONS.VOTING:
+        if (votingComplete) {
+          setMovingMapComplete(false);
+          setAction(ACTIONS.MOVING_MAP);
+        }
+        break;
+      case ACTIONS.MOVING_MAP:
+        if (movingMapComplete) {
+          setAction(ACTIONS.SOLVING);
+        }
+        break;
+      default:
+        // eslint-disable-next-line no-console
+        console.warn("Unknown action: ", action);
+        break;
+    }
+  }, [prevActiveTask, activeTask, votingComplete, movingMapComplete]);
+
+  // useEffect(() => {
+  //   if (activeTask && activeTask !== prevActiveTask) {
+  //     // eslint-disable-next-line no-console
+  //     console.log("new task");
+  //   }
+  // }, [activeTask, prevActiveTask]);
+
+  // when the user transitions from one action to another
+  useEffect(() => {
+    if (action !== prevAction) {
+      switch (action) {
+        case ACTIONS.SOLVING:
+          break;
+        case ACTIONS.VOTING:
+          startTimer(votingDuration);
+          break;
+        case ACTIONS.MOVING_MAP:
+          startTimer(mapTransitionDuration);
+          break;
+        default:
+          // eslint-disable-next-line no-console
+          console.log("unknown action ", action);
+      }
+    }
+  }, [action, prevAction]);
+
+  const onItemSelection = item => {
+    // eslint-disable-next-line no-console
+    console.log("item ", item);
+    // if (item.type === activeTask.requiredItem) {
+    //   this.setState(state => ({
+    //     correctItemsChosen: state.correctItemsChosen + 1
+    //   }));
+    // }
+  };
+
+  const onTaskSelection = task => {
+    // eslint-disable-next-line no-console
+    console.log("onTaskSelection ", task);
+    // const { taskVotes } = this.state;
+    // taskVotes[task.type] += 1;
+
+    // this.setState({
+    //   taskVotes
+    // });
+    // // Return true so Orb knows how to animate
+    // return true;
+  };
+
+  const isSolving = action === ACTIONS.SOLVING;
+  const possibleItems = isSolving ? playerKitItems : weightedTasks;
+  const frozenOrbInterface = !isSolving;
+  const onOrbSelection = isSolving ? onItemSelection : onTaskSelection;
 
-  return activeTask ? <SolveScreen /> : <ChooseScreen />;
+  return (
+    <Fragment>
+      <div css={mapAndInfoStyle}>
+        <SolveScreen activeTask={activeTask} open={isSolving} />
+        <ChooseScreen
+          activeTask={activeTask}
+          votingComplete={votingComplete}
+          movingMapComplete={movingMapComplete}
+        />
+        {debug && <TaskDebugger activeTask={activeTask} action={action} />}
+      </div>
+      <DurationBar
+        step="Choose a task"
+        debug
+        percentComplete={percentComplete}
+      />
```

Comment:
When ACTIONS.MOVING_MAP is true, we should not reset or hide this element

Reply:
Let's touch base about this post-merge.